x-pack/plugin/redact/src/main/java/org/elasticsearch/xpack/redact/RedactProcessor.java

```diff
@@ -203,6 +215,20 @@ private static void matchRepeat(Grok grok, byte[] utf8Bytes, RegionTrackingMatch
         } while (offset != length);
     }
 
+    private void updateMetadataIfNecessary(IngestDocument ingestDocument, String fieldValue, String redacted) {
+        if (traceRedact == false) return;
+        if (fieldValue == null) return;
+
+        Boolean isRedactedMetadata = ingestDocument.getFieldValue(METADATA_PATH_REDACT_IS_REDACTED, Boolean.class, true);
+        boolean alreadyRedacted = Boolean.TRUE.equals(isRedactedMetadata);
+        boolean isRedacted = fieldValue.equals(redacted) == false;
+
+        // document newly redacted
+        if (alreadyRedacted == false && isRedacted) {
```

Comment:
Does it matter if it's already redacted? We only skip setting this if the metadata field is true already. It doesn't look like it can ever be false. Couldn't we just overwrite the field regardless of its existing value?

Reply:
It is possible that a redact processor does not actually modify the document (the pattern is not found), in that case I think we should not set `_is_redacted` to true . 

In the case of multiple redact processors, if the first processor redacts the doc and sets `_is_redacted` to true, and the second processor does not redact the doc, we want to skip setting `_is_redacted` in the second processor to prevent overriding the previously true value.